Fix breakNaturally for fluid-logged blocks

Machine-Maker opened the pull request:
When a player breaks a fluid-logged block, the fluid remains. This is not currently the case for breakNaturally. I think this isn't a breaking change cause the method is **supposed** to simulate to the best it can, a block being broken by a player.

Owen1212055 reviewed (approved):
Can confirm works just fine when tested

Machine-Maker committed:
Fix fluid-logging on Block#breakNaturally

Leaves fluid if the block broken was fluid-logged which is what
happens if a player breaks a fluid-logged block
